fennel.lua

```diff
@@ -2456,11 +2476,11 @@ local replsource = [===[(local (fennel internals) ...)
   (let [matches []
         input-fragment (text:gsub ".*[%s)(]+" "")]
     (fn add-partials [input tbl prefix] ; add partial key matches in tbl
-      (each [k (pairs tbl)]
+      (each [k (internals.allPairs tbl)]
         (let [k (if (or (= tbl env) (= tbl env.___replLocals___))
                     (. scope.unmanglings k)
                     k)]
-          (when (and (< (# matches) 40)
+          (when (and (< (# matches) 2000) ; stop explosion on too many items
```

Comment:
I think you're right that we should keep the limit just in case; if anyone needs more than 2000 it's likely they're doing something so specialized that they can do this part for themselves.